packages/plugins/src/hardhat/tasks.ts

```diff
@@ -148,6 +158,177 @@ subtask(TASK_CHUGSPLASH_FETCH)
     }
   )
 
+// TODO: Change provider to a non-local RPC provider after demo
+task(TASK_CHUGSPLASH_INIT)
+  .setDescription('Creates a new ChugSplash project')
+  .addParam('deployConfig', 'path to chugsplash deploy config')
+  .setAction(
+    async (
+      args: {
+        deployConfig: string
+      },
+      hre
+    ) => {
+      spinner.start('Creating new project...')
+
+      const config: ChugSplashConfig = await hre.run(TASK_CHUGSPLASH_LOAD, {
+        deployConfig: args.deployConfig,
+      })
+
+      const provider = new providers.JsonRpcProvider()
+      const ChugSplashRegistry = new Contract(
+        CHUGSPLASH_REGISTRY_ADDRESS,
+        ChugSplashRegistryABI,
+        await provider.getSigner()
+      )
+
+      await ChugSplashRegistry.register(
+        config.options.name,
+        config.options.owner
+      )
+
+      spinner.succeed('Project successfully created')
+    }
+  )
+
+task(TASK_CHUGSPLASH_LIST_ALL_PROJECTS)
+  .setDescription('Lists all existing ChugSplash projects')
+  .setAction(async () => {
+    spinner.start('Getting list of all projects...')
+
+    const provider = new providers.JsonRpcProvider()
+    const ChugSplashRegistry = new Contract(
+      CHUGSPLASH_REGISTRY_ADDRESS,
+      ChugSplashRegistryABI,
+      provider
+    )
+
+    const events = await ChugSplashRegistry.queryFilter(
+      ChugSplashRegistry.filters.ChugSplashProjectRegistered()
+    )
+
+    spinner.stop()
+    events.forEach((event) =>
+      console.log(
+        `Project: ${event.args.nonIndexedProjectName}\t\tManager: ${event.args.manager}`
+      )
+    )
+  })
+
+task(TASK_CHUGSPLASH_PROPOSE)
+  .setDescription('Proposes a new ChugSplash bundle')
+  .addParam('bundleHash', 'hash of the bundle')
```

Comment:
Probably easier if we just call into `chugsplash-commit` to automatically generate this. Simplifies the UX a bit and it just becomes a single `chugsplash-propose` step instead of having to run `chugsplash-commit` first.